app/controllers/user_challenges_controller.rb

```diff
@@ -17,45 +18,56 @@ def create
   end
 
   def destroy
-    uc = UserChallenge.includes(:challenge).find_by id: params[:id]
-    challenge = uc.challenge
+    destroyer = UserChallenge::Destroyer.new(id: params[:id])
+    result = destroyer.destroy
 
-    if challenge.open_spot_challenge_type?
-      remove_user_from_open_spot_challenge uc
+    if result.success?
+      head :no_content
     else
-      remove_user_from_non_open_spot_challenge uc
+      render json: {
+        resource: 'user_challenge',
+        errors: [user_challenge: result.errors.first]
+      }, status: :unprocessable_entity
     end
   end
 
-  private
+  def evaluate_comments
+    evaluator = UserChallenge::Evaluator.new(params: params)
+    result = evaluator.save_comments
 
-  def remove_user_from_open_spot_challenge(user_challenge)
-    challenge = user_challenge.challenge
-    if challenge.users.length <= 1 && user_challenge.destroy && challenge.destroy
-      head 204
-    elsif user_challenge.destroy
-      head 204
+    if result.success?
+      head :no_content
     else
-      render json: { resource: 'user_challenge', errors: [challenge: 'could not be destroyed'] }, status: 500
+      render json: {
+        resource: 'user_challenge',
+        errors: [user_challenge: result.errors.first]
+      }, status: :unprocessable_entity
     end
   end
 
-  def remove_user_from_non_open_spot_challenge(user_challenge)
-    challenge = user_challenge.challenge
-    if challenge.destroy
-      head 204
+  def evaluate_places
+    evaluator = UserChallenge::Evaluator.new(params: params)
+    result = evaluator.save_places
+
+    if result.success?
+      head :no_content
     else
-      render json: { resource: 'user_challenge', errors: [challenge: 'could not be destroyed'] }, status: 500
+      render json: {
+        resource: 'user_challenge',
+        errors: [user_challenge: result.errors.first]
+      }, status: :unprocessable_entity
     end
   end
 
+  private
+
   def ensure_challenge_exists!
-    return unless Challenge.find_by(id: params[:challenge_id]).nil?
+    return if Challenge.exists? id: params[:challenge_id]
     render json: { resource: 'user_challenge', errors: [challenge: 'challenge doesn\'t exist'] }, status: 403
   end
 
   def ensure_user_exists!
-    return unless User.find_by buck_id: params[:buck_id].nil?
+    return if User.exists? buck_id: params[:challenger_buck_id]
```

Comment:
This was originally doing two things wrong:
```ruby
return unless User.find_by(buck_id: params[:buck_id].nil?)
```

`params[:challenger_buck_id]` is what you are actually looking for, but you were getting false positives because of the parenthesis issue